webui/react/src/pages/TrialDetails/TrialDetailsLogs.tsx

```diff
@@ -182,23 +223,220 @@ const TrialDetailsLogs: React.FC<Props> = ({ experiment, trial }: Props) => {
+                <LogViewerEntry
+                  formattedTime={logEntry.formattedTime}
+                  htmlMessage={true}
+                  key={logEntry.id}
+                  level={logEntry.level}
+                  message={logEntry.message}
+                  style={{
+                    backgroundColor:
+                      logEntry.id === selectedLog?.id ? 'var(--theme-ix-active)' : 'transparent',
+                  }}
+                />
+              </div>
+            ))
+          ) : (
+            <Message icon="warning" title="No logs to show. " />
+          )}
+        </div>
+      </div>
+    );
+  }, [settings.enableRegex, formattedSearchResults, container, selectedLog, updateSettings]);
+
+  const formatClipboardHeader = (log: Log): string => {
+    const logEntry = formatLogEntry(log);
+    const format = `%${MAX_DATETIME_LENGTH - 1}s `;
+    const level = `<${logEntry.level || ''}>`;
+    return sprintf(`%-9s ${format}`, level, logEntry.formattedTime);
+  };
+
+  const clipboardCopiedMessage = useMemo(() => {
+    const linesLabel = pluralizer(logs.length, 'entry', 'entries');
+    return `Copied ${logs.length} ${linesLabel}!`;
+  }, [logs]);
+
+  const getClipboardContent = useCallback(() => {
+    return logs.map((log) => `${formatClipboardHeader(log)}${log.message || ''}`).join('\n');
+  }, [logs]);
+
+  const handleFullScreen = useCallback(() => {
+    if (logsRef.current && screenfull.isEnabled) screenfull.toggle();
+  }, []);
+
+  const rightButtons = (
+    <Row>
+      <ClipboardButton copiedMessage={clipboardCopiedMessage} getContent={getClipboardContent} />
+      <Button
+        aria-label="Toggle Fullscreen Mode"
+        icon={<Icon name="fullscreen" showTooltip title="Toggle Fullscreen Mode" />}
+        onClick={handleFullScreen}
+      />
+      <Button
+        aria-label="Download Logs"
+        icon={<Icon name="download" showTooltip title="Download Logs" />}
+        onClick={handleDownloadLogs}
+      />
+    </Row>
+  );
+
   return (
-    <div className={css.base}>
+    <div className={css.base} ref={container}>
       <Spinner conditionalRender spinning={!trial}>
-        <LogViewer
-          decoder={mapV1LogsResponse}
-          serverAddress={serverAddress}
-          title={logFilters}
-          onDownload={handleDownloadLogs}
-          onError={handleError}
-          onFetch={handleFetch}
+        <div className={css.header}>
+          <div className={css.filters}>
+            <Input
+              allowClear
+              placeholder="Search Logs..."
+              value={searchInput || settings.searchText}
+              width={240}
+              onChange={onSearchChange}
+            />
+            <Tooltip content={`${searchOn ? 'Close' : 'Open'} Search`}>
+              <Button
+                type={searchOn ? 'primary' : 'default'}
+                onClick={() => {
+                  setSearchOn((prev) => !prev);
+                  searchOn && setLogViewerOn(true);
```

Comment:
consider swapping the lines here -- otherwise it looks like we're trying to use `searchOn`'s new value after setting it. We're not, of course, but i could see someone "helpfully" trying to fix this without understanding what's happening here.